### PR TITLE
Add documentation for using other models, make it easier to use non OpenAI models

### PR DESCRIPTION
**Pull request by PCSwingle**
Replaces #450. After discussing with @jakethekoenig, we feel this is a better alternative than using litellm directly in the codebase.

## Pull Request Checklist
- [x] Documentation has been updated, or this change doesn't require that

### CODE REVIEW

File: mentat/code_context.py [hunk 1]
```diff
@@ -148,12 +148,13 @@ async def get_code_message(
             "\n".join(include_files_message), model, full_message=False
         )
 
-        tokens_used = (
-            prompt_tokens + meta_tokens + include_files_tokens + config.token_buffer
-        )
+        tokens_used = prompt_tokens + meta_tokens + include_files_tokens
         if not is_context_sufficient(tokens_used):
-            raise ContextSizeInsufficient()
```
Comment (jakethekoenig) [inline mentat/code_context.py:155]: Should we print something here so the user knows what the problem is?

File: README.md [hunk 2]
```diff
@@ -70,9 +74,35 @@ You'll need to have API access to GPT-4 to run Mentat. There are a few options t
 
 Mentat also works with the Azure OpenAI API. To use the Azure API, provide the `AZURE_OPENAI_ENDPOINT` (`https://<your-instance-name>.openai.azure.com/`) and `AZURE_OPENAI_KEY` environment variables instead of `OPENAI_API_KEY`.
 
-In addition, Mentat uses the `gpt-4-1106-preview` by default. On Azure, this model is available under a different name: `gpt-4-1106-Preview` (with a capital P). To use it, override the default model as described in [configuration.md](docs/configuration.md).
+In addition, Mentat uses the `gpt-4-1106-preview` model by default. When using Azure, you will have to set the model as described in [configuration.md](docs/configuration.md) to the name you gave your Azure model.
```
Comment (jakethekoenig) [inline README.md:77]: We should duplicate this to the user guide and remove it from the user guide when that's hosted. This is good for now though.

**Review by jakethekoenig — APPROVED** [review]
Looks good to me. Thanks!

File: mentat/code_context.py (hunk 1 above)
Comment (PCSwingle) [reply mentat/code_context.py:155]: We do print something in is_context_sufficient; maybe we should move the warning out of the function, but we do use it in multiple places.

**Pull request merged**